File: Easy/Reverse Coding/reverse-coding.py
#User function Template for python3

class Solution:
    def sumOfNaturals(self, n):
        mod = 1e9 + 7
        tsum = (n * (n + 1)) // 2
        
        return int(tsum % mod)


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__': 
    t = int (input ())
    for _ in range (t):
        n = int(input())
        
        ob = Solution()
        print(ob.sumOfNaturals(n))
# } Driver Code Ends

File: Easy/Reverse Coding/reverse-coding.js
//{ Driver Code Starts
//Initial Template for javascript

'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.trim().split('\n').map(string => {
        return string.trim();
    });

    main();
});

function readLine() {
    return inputString[currentLine++];
}


function main() {
    let t = parseInt(readLine());
    let i = 0;
    for (; i < t; i++) {
        let [n, m] = readLine().trim().split(" ").map((x) => parseInt(x));
        let obj = new Solution();
        let res = obj.revCoding(n, m);
        console.log(res);
    }
}
// } Driver Code Ends


//User function Template for javascript

/**
 * @param {number} n
 * @param {number} m
 * @return {number}
 */
class Solution
{
    revCoding(n, m)
    {
        const mod = 1e9 + 7;
        const x = (n * (n + 1)) / 2;
        
        return (x % mod);
    }
}